File: samples/client/3_0_3_unit_test/java/src/test/java/org/openapijsonschematools/client/components/schemas/AdditionalpropertiesCanExistByItselfTest.java
package org.openapijsonschematools.client.components.schemas;

import org.junit.Assert;
import org.junit.Test;
import org.openapijsonschematools.client.configurations.JsonSchemaKeywordFlags;
import org.openapijsonschematools.client.configurations.SchemaConfiguration;
import org.openapijsonschematools.client.exceptions.ValidationException;
import org.openapijsonschematools.client.schemas.validation.JsonSchemaFactory;
import org.openapijsonschematools.client.schemas.MapMaker;

import java.util.Arrays;
import java.util.List;
import java.util.AbstractMap;

public class AdditionalpropertiesCanExistByItselfTest {
    static final SchemaConfiguration configuration = new SchemaConfiguration(JsonSchemaKeywordFlags.ofNone());
    static final AdditionalpropertiesCanExistByItself.AdditionalpropertiesCanExistByItself1 schema = JsonSchemaFactory.getInstance(
        AdditionalpropertiesCanExistByItself.AdditionalpropertiesCanExistByItself1.class
    );

    @Test
    public void testAnAdditionalValidPropertyIsValidPasses() {
        // an additional valid property is valid
        schema.validate(
            MapMaker.makeMap(
                new AbstractMap.SimpleEntry<>(
                    "foo",
                    true
                )
            ),
            configuration
        );
    }

    @Test
    public void testAnAdditionalInvalidPropertyIsInvalidFails() {
        // an additional invalid property is invalid
        Assert.assertThrows(ValidationException.class, () -> schema.validate(
            MapMaker.makeMap(
                new AbstractMap.SimpleEntry<>(
                    "foo",
                    1
                )
            ),
            configuration
        ));
    }
}

File: samples/client/3_0_3_unit_test/kotlin/src/test/kotlin/org/openapijsonschematools/client/components/schemas/AdditionalpropertiesCanExistByItselfTest.kt
package org.openapijsonschematools.client.components.schemas

import org.openapijsonschematools.client.configurations.JsonSchemaKeyword
import org.openapijsonschematools.client.configurations.SchemaConfiguration
import org.openapijsonschematools.client.exceptions.ValidationException

import kotlin.test.Test
import kotlin.test.assertFailsWith

class AdditionalpropertiesCanExistByItselfTest {
    companion object {
        val configuration = SchemaConfiguration(disabledKeywordFlags=setOf(JsonSchemaKeyword.FORMAT))
    }

    @Test
    fun testAnAdditionalValidPropertyIsValidPasses() {
        // an additional valid property is valid
        val schema = AdditionalpropertiesCanExistByItself.AdditionalpropertiesCanExistByItself1.getInstance()
        schema.validate(
            mapOf(
                Pair(
                    "foo",
                    true
                )
            ),
            configuration
        )
    }

    @Test
    fun testAnAdditionalInvalidPropertyIsInvalidFails() {
        // an additional invalid property is invalid
        val schema = AdditionalpropertiesCanExistByItself.AdditionalpropertiesCanExistByItself1.getInstance()
        assertFailsWith<ValidationException>(
            block = {
                schema.validate(
                    mapOf(
                        Pair(
                            "foo",
                            1
                        )
                    ),
                    configuration
                )
            }
        )
    }
}
